fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -281,8 +283,8 @@ private void prepare(
         context.setQueryTables(queryTables);
         context.setUpdateTableId(updateTableId);
 
-        // prepare related mvs if needed
-        new MvRewritePreprocessor(connectContext, columnRefFactory,
+        // prepare related mvs if needed and initialize mv rewrite strategy
+        mvRewriteStrategy = new MvRewritePreprocessor(connectContext, columnRefFactory,
```

Comment:
It seems that mvRewriteStrategy can be decide without MvRewritePreprocessor?
better to split it into two steps:
1. use MvStrategyArbitrator to decide the mvRewriteStrategy
2. use mvRewriteStrategy to in MvRewritePreprocessor.prepare to do preparation.

Reply:
done